IDL/maps.idl

```diff
@@ -0,0 +1,269 @@
+#include "helpers/basic_inner_types.idl"
+
+typedef short alias_array[2];
+typedef sequence<short> alias_sequence;
+typedef map<long, long> alias_map;
+
+struct MapShort
+{
+    map<long, short> var_map_short;
+};
+
+struct MapUShort
+{
+    map<long, unsigned short> var_map_ushort;
+};
+
+struct MapLong
+{
+    map<long, long> var_map_long;
+};
+
+struct MapULong
+{
+    map<long, unsigned long> var_map_ulong;
+};
+
+struct MapLongLong
+{
+    map<long, long long> var_map_longlong;
+};
+
+struct MapULongLong
+{
+    map<long, unsigned long long> var_map_ulonglong;
+};
+
+struct MapFloat
+{
+    map<long, float> var_map_float;
+};
+
+struct MapDouble
+{
+    map<long, double> var_map_double;
+};
+
+struct MapLongDouble
+{
+    map<long, long double> var_map_longdouble;
+};
+
+struct MapBoolean
+{
+    map<long, bool> var_map_boolean;
+};
+
+struct MapOctet
+{
+    map<long, octet> var_map_octet;
+};
+
+struct MapChar
+{
+    map<long, char> var_map_char;
+};
+
+struct MapWChar
+{
+    map<long, wchar> var_map_wchar;
+};
+
+struct MapString
+{
+    map<long, string> var_map_string;
+};
+
+struct MapWString
+{
+    map<long, wstring> var_map_wstring;
+};
+
+struct MapEnum
+{
+    map<long, MyEnum> var_map_enum;
+};
+
+struct MapBitMask
+{
+    map<long, MyBitMask> var_map_bitmask;
+};
+
+struct MapAlias
+{
+    map<long, MyAlias> var_map_alias;
+};
+
+struct MapShortArray
+{
+    map<long, alias_array> var_map_short_array;
+};
+
+struct MapSequence
+{
+    map<long, alias_sequence> var_map_sequence;
+};
+
+struct MapMap
+{
+    map<long, alias_map> var_map_map;
+};
+
+struct MapUnion
+{
+    map<long, MyUnion> var_map_union;
+};
+
+struct MapStructure
+{
+    map<long, MyStructure> var_map_structure;
+};
+
+struct MapBitset
+{
+    map<long, MyBitset> var_map_bitset;
+};
+    
+
+
+struct MapKeyShort
+{
+    std::map<short, short> var_map_short;
+};
+
+struct MapKeyUShort
+{
+    std::map<unsigned short, unsigned short> var_map_ushort;
+};
+
+struct MapKeyLong
+{
+    std::map<long, long> var_map_long;
+};
+
+struct MapKeyULong
+{
+    std::map<unsigned long, unsigned long> var_map_ulong;
+};
+
+struct MapKeyLongLong
+{
+    std::map<long long, long long> var_map_longlong;
+};
+
+struct MapKeyULongLong
+{
+    std::map<unsigned long long, unsigned long long> var_map_ulonglong;
+};
+
+struct MapKeyFloat
+{
+    std::map<float, float> var_map_float;
+};

```

Comment:
From XTYPES v1.3 specification:

> Map types have an additional attribute, the key element type, that indicates the type of the key elements. Implementers of this specification need only support key elements of signed and unsigned integer types and of narrow and wide string types; the behavior of maps with other key element types is undefined and may not be portable. (Design rationale, non-normative: Support for arbitrary key element types would require implementers to provide uniform sorting and/or hashing operations, which would be impractical on many platforms. In contrast, these operations have straightforward implementations for integer and string types.)

Several map keys should be removed according to the specification  (or at least that is the minimal support required).

Reply:
Removed in [e3abe74](https://github.com/eProsima/dds-types-test/pull/1/commits/e3abe748b33c5d42e70f957439410b6722e52980)